internal/handler/handler.go

```diff
@@ -105,16 +76,93 @@ func (ch *CommandHandler) Handle(client i.ClientInterface, command string) bool
 	return flag
 
 }
-func (cmdH *CommandHandler) InitCommandHandler() i.CommandHandlerInterface {
+func (cmdH *CommandHandler) InitCommandHandler() common.CommandHandlerInterface {
 
-	cmdH.RegisterCommand("^/join$", 0, JoinChat)
+	cmdH.RegisterCommand("^/join (?P<name>[a-zA-Z0-9]+)$", 1, JoinChat)
 	cmdH.RegisterCommand("^/nick (?P<name>[a-zA-Z0-9]+)$", 1, HandleChangeNickname)
 	cmdH.RegisterCommand("^/quit$", 0, QuitChat)
 	cmdH.RegisterCommand("^/whisper (?P<target>[a-zA-Z0-9]+) (?P<message>.+)$", 2, HandleWhisper)
 	cmdH.RegisterCommand("^/users$", 0, GetNumberOfUsers)
 	cmdH.RegisterCommand("^/whois (?P<target>[a-zA-Z0-9]+)$", 1, Whois)
+	cmdH.RegisterCommand("^/register (?P<password>[a-zA-Z0-9]+)$", 1, HandleRegister)
+	cmdH.RegisterCommand("^/login (?P<user>[a-zA-Z0-9]+) (?P<password>[a-zA-Z0-9]+)$", 2, HandleLogin)
 
 	return cmdH
 
 }
 
+func HandleRegister(ctx common.Context) {
+	username := ctx.Client.GetUsername()
+
+	if username == "Server" {
+		ctx.Server.SendServerPrivateMessage(fmt.Sprintf(common.REGISTER_FAILED, username), ctx.Client)
+		return
+	}
+
+	if ctx.Server.IsUserRegistered(username) {
+		ctx.Server.SendServerPrivateMessage(fmt.Sprintf(common.USER_ALREADY_REGISTERED, username), ctx.Client)
+		return
+	}
+
+	ctx.Server.RegisterUser(username, ctx.Get("password"))
+
+}
+
+func HandleLogin(ctx common.Context) {
+	if ctx.Server.IsUserCredentialsValid(ctx.Get("user"), ctx.Get("password")) {
+		ctx.Client.SetUsername(ctx.Get("user"))
+		ctx.Server.BroadcastServerMessage(fmt.Sprintf(common.LOGIN_SUCCESS, ctx.Get("user")))
```

Comment:
should we add the user already in chat logic here as well ?
cause you could login from another client when user is already actively logged into  a client.

Reply:
that's technically ok for now? isn't it?